### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**
[DEMO LINK](https://andriy-shymkiv.github.io/react_clock/)

### CODE REVIEW

**Review by Artem66 — APPROVED** [review]
as for me everything is fine

File: src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.addEventListener('click', (event) => {
+      if (event) {
+        this.setState({ hasClock: true });
+      }
+    });
```
Comment (denys-danyliuk) [inline src/App.tsx:39]: move callback to method, watch this video from 5:05 to know why
https://mate.academy/learn/react/class-components#/video/3221

File: src/App.tsx [hunk 2]
```diff
@@ -1,39 +1,56 @@
...
+  }
 
-        {' time is '}
+  componentWillUnmount() {
+    clearInterval(this.timerId);
+  }
```
Comment (denys-danyliuk) [inline src/App.tsx:44]: you also need to unmount your event listeners

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

**Review by Olexandr6 — APPROVED** [review]
Good job 👍

**Pull request closed without merging**